### ISSUE TITLE
Collection variable not interpolated in GraphQL vars

### ISSUE THREAD

**Issue by rbideau** [usebruno/bruno#2960]
### I have checked the following:

- [X] I use the newest version of bruno.
- [X] I've searched existing issues and found nothing related to my issue.

### Describe the bug

On version 1.27.0, collection variable are not interpolated in GraphQL Vars when executing the request via the app.

Might be related to

- #352 / #1217 
- #2871 

### .bru file to reproduce the bug

```
meta {
  name: spacex
  type: graphql
  seq: 1
}

post {
  url: https://spacex-production.up.railway.app/
  body: graphql
  auth: none
}

body:graphql {
  {
    capsules {
      id
    }
  }
  
}

vars:post-response {
  id: res.body.data.capsules[0].id
}

assert {
  res.status: eq 200
}
```

```
meta {
  name: spacex var
  type: graphql
  seq: 2
}

post {
  url: https://spacex-production.up.railway.app/
  body: graphql
  auth: none
}

body:graphql {
  query ($id: ID!) {
    capsule(id: $id) {
      id
      landings
    }
  }
  
}

body:graphql:vars {
  {"id": "{{id}}"}
}

assert {
  res.status: eq 200
}
```

### Screenshots/Live demo link

[Screencast from 29-08-2024 08:35:56.webm](https://github.com/user-attachments/assets/ed6e6009-7f75-462f-b613-87a6469b559d)

**Comment by mkurapov** [comment]
Not sure if this is something the Bruno team will be looking to add in the future, but in the meantime @rbideau, we've been using this:

```
  resolveTemplateVariables: function (string) {
    const VARIABLE_NAME_REGEX = /{{([A-Za-z]\w+)}}/g

    return string.replace(
      VARIABLE_NAME_REGEX,
      (_, key) => bru.getVar(key) || bru.getEnvVar(key)
    )
  }
```

**Comment by mkurapov** [comment]
Ah, never mind, I see this is a net-new/separate issue. I think we needed to manually interpolate the string in the query, but never in the input variables

**Comment by Nightbr** [comment]
Still on v1.28.0

Also it was working before since we have many request that uses it.

Happy to help debugging or giving some tests example if necessary.

**Comment by ben-sai** [comment]
This is a regression and worked fine in 1.24 I had to go back to earlier build

**Comment by cmuench** [comment]
I was able to get it working by using the script of @mkurapov in a collection pre-script a bit changed:

```js
function resolveTemplateVariables(string) {
  const VARIABLE_NAME_REGEX = /{{([A-Za-z_]\w+)}}/g;
  
  return string.replace(
    VARIABLE_NAME_REGEX,
    (_, key) => bru.getVar(key) || bru.getEnvVar(key)
  );
}

let body = req.getBody();

if (!body.variables) {
  return;
}

let resolvedVariables = resolveTemplateVariables(body.variables);
body.variables = resolvedVariables;
//console.log(body);
```

Works well with my Magento 2 project.

**Comment by V-ed** [comment]
I did something similar but kept copying it in every request, thank you @cmuench for making me realize that indeed the collections scripts exists!

Although your script is fairly complete, there's two things I changed in my version :

```javascript
// Bruno apparently does not support variables substitution in GQL queries yet :)
// Doing this temporarily until this is fixed.
// https://github.com/usebruno/bruno/issues/2960

let body = req.getBody();

if (!body?.variables) {
  return;
}

function resolveTemplateVariables(string) {
  const VARIABLE_NAME_REGEX = /{{(\w+)}}/g;
  
  return string.replace(
    VARIABLE_NAME_REGEX,
    (_, key) => bru.getVar(key) || bru.getEnvVar(key)
  );
}

const resolvedVariables = resolveTemplateVariables(body.variables);

body.variables = resolvedVariables;
```

1. The variable `body` could be `undefined` in regular queries / when there is no Body, so adding a shortcircuit helps prevent weird `TypeError: Cannot read properties of undefined (reading 'variables')` errors in non-graphql requests
2. Tweaked the `VARIABLE_NAME_REGEX`, as the only important bits of the regex is the `\w`. I know this was copied from @mkurapov's comment, no diss to either of you, it just makes the intent of this regex clearer (unless you REALLY wanted your variables to start by a letter)!

Hopefully this can get fixed so we don't need this workaround (especially since it seems like it was a regression, having hover info of the variable value in the GQL variables view would be super helpful!).

**Comment by mjhcorporate** [comment]
The interpolation is only broken in the UI -- it works on the command line. However, the pre-request script breaks the CLI usage, because on the CLI, `body.variables` is an object!

Revised pre-request script checks for this:
```js
// Bruno has a regression w.r.t. environment variables in GraphQL variables.
// Doing a manual replacement until the issue is fixed.
// https://github.com/usebruno/bruno/issues/2960

let body = req.getBody();

if (!body?.variables || (typeof body.variables === 'object')) {
  return;
}

function resolveTemplateVariables(string) {
  const VARIABLE_NAME_REGEX = /{{(\w+)}}/g;
  return string.replace(
    VARIABLE_NAME_REGEX,
    (_, key) => bru.getVar(key) || bru.getEnvVar(key)
  );
}

const resolvedVariables = resolveTemplateVariables(body.variables);

body.variables = resolvedVariables;
```

**Comment by srpalwaidynamite** [comment]
Hopefully they fix this issue in new version

**Comment by rbideau** [comment]
I can no longer reproduce it on [1.32.0](https://github.com/usebruno/bruno/releases/tag/v1.32.0)

Looks like it was fixed by #3212
